test/services/WelshTextBuilderSpec.scala

```diff
@@ -54,185 +48,106 @@ class TextBuilderSpec extends BaseSpec {
+      TextBuilder.fromPhrase(p).items shouldBe Seq(Words("Sentence with a "), LabelRef("BLAH", Txt, true), Words(" label reference"))
     }
 
     "Convert a label placeholderwith currency output format within a bold placeholder to a bold label ref" in new Test {
       val p = Phrase("""Sentence with a [bold:[label:BLAH:currencyPoundsOnly]] label reference""", """Sentence with a [bold:[label:BLAH]:currencyPoundsOnly]] label reference""")
-
-      val txt = TextBuilder.fromPhrase(p)
-      txt.english shouldBe Seq(Words("Sentence with a "), LabelRef("BLAH", CurrencyPoundsOnly, true), Words(" label reference"))
+      TextBuilder.fromPhrase(p).items shouldBe Seq(Words("Sentence with a "), LabelRef("BLAH", CurrencyPoundsOnly, true), Words(" label reference"))
     }
 
     "Convert button link placeholders within phrase to Link as button TextItems" in new Test {
-
       val p = Phrase("""Sentence with a [button:BLAH:3] label reference""", """Sentence with a [button:BLAH:3] label reference""")
-
-      val txt = TextBuilder.fromPhrase(p)(urlMap1)
-
-      txt.english shouldBe Seq(Words("Sentence with a "), Link("dummy-path", "BLAH", false, true), Words(" label reference"))
-      txt.welsh shouldBe Seq(Words("Sentence with a "), Link("dummy-path", "BLAH", false, true), Words(" label reference"))
-
+      TextBuilder.fromPhrase(p)(urlMap1, lang).items shouldBe Seq(Words("Sentence with a "), Link("dummy-path", "BLAH", false, true), Words(" label reference"))
     }
 
     "Convert button link in tab placeholders within phrase to Link as button TextItems" in new Test {
-
       val p = Phrase("""Sentence with a [button-tab:BLAH:3] label reference""", """Sentence with a [button-tab:BLAH:3] label reference""")
-
-      val txt = TextBuilder.fromPhrase(p)(urlMap1)
-
-      txt.english shouldBe Seq(Words("Sentence with a "), Link("dummy-path", "BLAH", true, true), Words(" label reference"))
-      txt.welsh shouldBe Seq(Words("Sentence with a "), Link("dummy-path", "BLAH", true, true), Words(" label reference"))
-
+      TextBuilder.fromPhrase(p)(urlMap1, lang).items shouldBe Seq(Words("Sentence with a "), Link("dummy-path", "BLAH", true, true), Words(" label reference"))
     }
 
     "Convert a Text with link placeholders in lang strings to Seq[TextItem]" in new Test {
-
       val txt = TextBuilder.fromPhrase(txtWithLinks)
-
-      txt.english(0) shouldBe lEnWords1
-      txt.welsh(0) shouldBe lCyWords1
-
-      txt.english(1).toWords shouldBe Link("https://www.bbc.co.uk", link1EnWords).toWords
-      txt.welsh(1).toWords shouldBe Link("https://www.bbc.co.uk", link1CyWords).toWords
-
-      txt.english(2) shouldBe lEnWords2
-      txt.welsh(2) shouldBe lCyWords2
-
-      txt.english(3).toWords shouldBe Link("https://www.gov.uk", link2EnWords).toWords
-      txt.welsh(3).toWords shouldBe Link("https://www.gov.uk", link2CyWords).toWords
-
-      txt.english(4) shouldBe lEnWords3
-      txt.welsh(4) shouldBe lCyWords3
+      txt.items(0) shouldBe words1
```

Comment:
This test will fail when lang equals "cy"

Reply:
fixed